### PR TITLE
InstagramPhoto: Add blurred border option

### PR DESCRIPTION
**Pull request by RainGrid**
The possibility of using the photo as a blurred border during the operation of expansion

### CODE REVIEW

**Comment by RainGrid** [comment]
Possibly need to add:
- strength of the filter
- scale factor for the blur image
And I am not sure about the rotation operation, the background color wiil be applied for both blurred and non-blurred borders if the dimensions change.

**Comment by JRoy** [comment]
I am not able to trigger this. Could you please leave a sample photo and sample code you used.

**Comment by RainGrid** [comment]
@JRoy Code example:
```php
set_time_limit(0);
date_default_timezone_set('UTC');
require __DIR__.'/../vendor/autoload.php';
$photoFilename = __DIR__ . '/flower.jpg';
$captionText = 'cap';
try {
    $photo = new \InstagramAPI\Media\Photo\InstagramPhoto($photoFilename, array('useRecommendedRatio' => true, 'operation' => 2, 'bgColor' => array(0, 200, 200), 'blurredBorder' => true));
    $file = $photo->getFile();
    $data = file_get_contents($file);
    file_put_contents(__DIR__. '/testPhotoResult.jpg', $data);
} catch (\Exception $e) {
    echo 'Something went wrong: '.$e->getMessage()."\n";
}
```
Photos:
testPhotoResult
testPhotoResult4
testPhotoResult2

**Comment by mgp25** [comment]
@RainGrid haven't seen the code yet but the result looks awesome.

Great job! 👍

File: src/Media/Photo/InstagramPhoto.php [hunk 1]
```diff
@@ -161,8 +170,49 @@ protected function _processResource(
...
+            }
+
+            // Copy the image to the scaled canvas
+            if (imagecopyresampled(
+                $scaledDownImage, $source,
+                0, 0,
+                $blurImageRect->getX(), $blurImageRect->getY(),
+                $scaleDownRect->getWidth(), $scaleDownRect->getHeight(),
+                $blurImageRect->getWidth(), $blurImageRect->getHeight()
+            ) === false) {
```
Comment (JRoy) [inline src/Media/Photo/InstagramPhoto.php]: The `) === false) {` part should not be its own line. It should join the last line.

File: src/Media/Photo/InstagramPhoto.php [hunk 2]
```diff
@@ -161,8 +170,49 @@ protected function _processResource(
...
+            }
+
+            //Copy the blurred image to the output canvas.
+            if (imagecopyresampled(
+                $output, $scaledDownImage,
+                0, 0,
+                0, 0,
+                $canvas->getWidth(), $canvas->getHeight(),
+                $scaleDownRect->getWidth(), $scaleDownRect->getHeight()
+            ) === false) {
```
Comment (JRoy) [inline src/Media/Photo/InstagramPhoto.php]: The `) === false) {` part should not be its own line. It should join the last line.

**Comment by JRoy** [comment]
Besides my requested changes, this PR is functional and the result is amazing indeed.

**Comment by RainGrid** [comment]
@JRoy I understand the required changes and will fix asap. I used the style of the whole class.

**Comment by JRoy** [comment]
No worries, I'm going to wait for @mgp25, to look this over but you have my approval!

**Comment by mgp25** [comment]
@JRoy @RainGrid looks fine!

**Comment by JRoy** [comment]
Thanks for your contribution!

**Pull request merged**